docs/java-rest/low-level/usage.asciidoc

```diff
@@ -65,6 +65,82 @@ http://hc.apache.org/httpcomponents-asyncclient-dev/[Apache Http Async Client]
 - commons-codec:commons-codec
 - commons-logging:commons-logging
 
+[[java-rest-low-usage-shading]]
+=== Shading
+
+In order to avoid version conflicts, the dependencies can be shaded and packaged
+within the client in a single JAR file (sometimes called "uber jar" or "fat
+jar"). Shading a dependency consists of taking its content (resources files and
+Java class files), rename its dependency packages before putting them in the
+same JAR file as the low-level Java REST client. Shading a jar can be
+accomplished by 3rd party plugins for gradle and maven.
+
+Be advised that shading a jar also has implications. Shading the commons logging
+layer, for instance, means that 3rd party logging backends need to be shaded as
+well.
+
+[[java-rest-low-usage-shading-maven]]
+==== Maven Sample
```

Comment:
I wonder if this should `float` so it's on the same page as the remainder of the content?

Reply:
One example is `Maven Sample` and the other is `Gradle configuration`; pick a common term?

Reply:
WRT `float`, It looks like 4-deep `====` does not create pages, an example is right above this in the `Maven` section, 

https://www.elastic.co/guide/en/elasticsearch/client/java-rest/5.6/java-rest-low-usage-maven.html

You do, however *have* to `[float]` anything thats 3 `===` deep to keep it on the same rendered page.

Reply:
Good catch, thanks for the clarification.